### PR TITLE
Fix PMD Warnings

### PR DESCRIPTION
**Pull request by KrisLimbo**
Fixed warnings in Jenkins job.

### CODE REVIEW

File: com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/util/JobChangeListener.java [hunk 1]
```diff
@@ -58,10 +58,7 @@ public synchronized boolean isJobDone(final String jobName) {
    */
   public synchronized boolean isJobSuccessful(final String jobName) {
     IStatus status = completedJobs.get(jobName);
-    if (status != null) {
-      return status.isOK();
-    }
-    return false;
+    return (status != null) && status.isOK();
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/util/JobChangeListener.java:61]:
```suggestion
    return status != null && status.isOK();
```

File: com.avaloq.tools.ddk.test.ui/src/com/avaloq/tools/ddk/test/ui/swtbot/SwtWorkbenchBot.java [hunk 2]
```diff
@@ -236,10 +236,8 @@ public boolean test() {
           @Override
           public Boolean run() {
             SWTBotShell wizardShell = activeShell();
-            if (wizardShell.widget.getData() instanceof WizardDialog) {
-              return ((WizardDialog) wizardShell.widget.getData()).getCurrentPage().getTitle().equals(wizardPageTitle);
-            }
-            return false;
+            return (wizardShell.widget.getData() instanceof WizardDialog)
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.test.ui/src/com/avaloq/tools/ddk/test/ui/swtbot/SwtWorkbenchBot.java:239]:
```suggestion
            return wizardShell.widget.getData() instanceof WizardDialog dialog

```

File: com.avaloq.tools.ddk.test.ui/src/com/avaloq/tools/ddk/test/ui/swtbot/SwtWorkbenchBot.java [hunk 3]
```diff
@@ -236,10 +236,8 @@ public boolean test() {
           @Override
           public Boolean run() {
             SWTBotShell wizardShell = activeShell();
-            if (wizardShell.widget.getData() instanceof WizardDialog) {
-              return ((WizardDialog) wizardShell.widget.getData()).getCurrentPage().getTitle().equals(wizardPageTitle);
-            }
-            return false;
+            return (wizardShell.widget.getData() instanceof WizardDialog)
+                && ((WizardDialog) wizardShell.widget.getData()).getCurrentPage().getTitle().equals(wizardPageTitle);
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.test.ui/src/com/avaloq/tools/ddk/test/ui/swtbot/SwtWorkbenchBot.java:240]:
```suggestion
                && dialog.getCurrentPage().getTitle().equals(wizardPageTitle);

```

File: com.avaloq.tools.ddk.xtext.format/src/com/avaloq/tools/ddk/xtext/format/resource/FormatResourceDescriptionStrategy.java [hunk 4]
```diff
@@ -88,11 +88,12 @@ public boolean createEObjectDescriptions(final EObject eObject, final IAcceptor<
...
-    if (!(eObject.toString().contains("self") || eObject.toString().contains("config") || eObject.toString().contains("elements")
-        || eObject.toString().contains("object"))) {
-      return super.createEObjectDescriptions(eObject, acceptor);
-    }
-    return false;
+    return shouldCreateDescriptionForObject(eObject) && super.createEObjectDescriptions(eObject, acceptor);
+  }
+
+  private boolean shouldCreateDescriptionForObject(final EObject eObject) {
+    return (!(eObject.toString().contains("self") || eObject.toString().contains("config") || eObject.toString().contains("elements")
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.format/src/com/avaloq/tools/ddk/xtext/format/resource/FormatResourceDescriptionStrategy.java:95]: could we refactor `eObject.toString()` to a variable?

**Review by rubenporras — CHANGES_REQUESTED** [review]

File: com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/util/JobChangeListener.java (hunk 1 above)
Comment (KrisLimbo) [reply com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/util/JobChangeListener.java]: ok. thank you.

File: com.avaloq.tools.ddk.test.ui/src/com/avaloq/tools/ddk/test/ui/swtbot/SwtWorkbenchBot.java (hunk 2 above)
Comment (KrisLimbo) [reply com.avaloq.tools.ddk.test.ui/src/com/avaloq/tools/ddk/test/ui/swtbot/SwtWorkbenchBot.java]: ok. thank you.

File: com.avaloq.tools.ddk.xtext.format/src/com/avaloq/tools/ddk/xtext/format/resource/FormatResourceDescriptionStrategy.java (hunk 4 above)
Comment (KrisLimbo) [reply com.avaloq.tools.ddk.xtext.format/src/com/avaloq/tools/ddk/xtext/format/resource/FormatResourceDescriptionStrategy.java]: ok. thank you.

File: com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/TestLabelFilter.java [hunk 5]
```diff
@@ -28,10 +28,7 @@ public TestLabelFilter(final String label) {
   @Override
   public boolean shouldRun(final Description description) {
     TestLabels labels = description.getAnnotation(TestLabels.class);
-    if (labels != null) {
-      return ArrayUtils.contains(labels.value(), label);
-    }
-    return false;
+    return (labels != null) && ArrayUtils.contains(labels.value(), label);
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/TestLabelFilter.java:31]:
```suggestion
    return labels != null && ArrayUtils.contains(labels.value(), label);
```

File: com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/junit/runners/TestLabelFilter.java [hunk 6]
```diff
@@ -30,10 +30,7 @@ public TestLabelFilter(final String label) {
   @Override
   public boolean shouldRun(final Description description) {
     TestLabels labels = description.getAnnotation(TestLabels.class);
-    if (labels != null) {
-      return ArrayUtils.contains(labels.value(), label);
-    }
-    return false;
+    return (labels != null) && ArrayUtils.contains(labels.value(), label);
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/junit/runners/TestLabelFilter.java:33]:
```suggestion
    return labels != null && ArrayUtils.contains(labels.value(), label);
```

**Review by rubenporras — CHANGES_REQUESTED** [review]

**Review by rubenporras — APPROVED** [review]

**Pull request merged**